lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -32,14 +32,18 @@ import java.time.LocalDate
 import java.time.format.DateTimeFormatter.*
 import java.time.format.DateTimeParseException
 
+/**
+ * Regex pattern to match date strings of the format yyyy-MM-dd
+ */
+@JvmField val DATE_PATTERN_REGEX = Regex("\\d\\d\\d\\d-\\d\\d-\\d\\d")
```

Comment:
Not sure why this is `@JvmField`?  That's only needed if you intend this to be accessed by Java clients.  Additionally, this should be `private` since we don't want this to be part of the API.

Reply:
If this is to be kept `private`, isn't it better to keep it as a `private` member inside the class?

Reply:
I don't know why it would be better.  I suppose that one could make an argument that we should avoid polluting the top-level namespace, but I don't think that applies here since there aren't a lot of top-level definitions in this file, and `private` means its not accessible to code in other files.